es6-shim.js

```diff
@@ -2097,6 +2170,233 @@
+          return true;
+        }
+
+        return false;
+      };
+
+      var willCreateCircularPrototype = function (object, proto) {
+        while (proto) {
+          if (object === proto) {
+            return true;
+          }
+
+          proto = Reflect.getPrototypeOf(proto);
+        }
+
+        return false;
+      };
+
+      // Some Reflect methods are basically the same as
+      // those on the Object global, except that a TypeError is thrown if
+      // target isn't an object. As well as returning a boolean indicating
+      // the success of the operation.
+      defineProperties(Reflect, {
+
+        // Apply method in a functional form.
+        apply: ES.Call,
+
+        // New operator in a functional form.
+        construct: function construct(constructor, args) {
+          if (!ES.IsCallable(constructor)) {
+            throw new TypeError('First argument must be callable.');
+          }
+
+          return ES.Construct(constructor, args);
+        },
+
+        defineProperty: wrapObjectFunction(Object.defineProperty),
+
+        // When deleting a non-existant or configurable property,
+        // true is returned.
+        // When attempting to delete a non-configurable property,
+        // it will return false.
+        deleteProperty: throwUnlessTargetIsObject(function deleteProperty(target, key) {
+          var desc = Object.getOwnPropertyDescriptor(target, key);
+
+          if (desc && !desc.configurable) {
+            return false;
+          }
+
+          // Will return true.
+          return delete target[key];
+        }),
+
+        enumerate: throwUnlessTargetIsObject(function enumerate(target) {
+          return new ObjectIterator(target, 'key');
+        }),
+
+        // Syntax in a functional form.
+        get: throwUnlessTargetIsObject(function get(target, key) {
+          var receiver = arguments.length > 2 ? arguments[2] : target;
+
+          return internal_get(target, key, receiver);
+        }),
+
+        getOwnPropertyDescriptor: throwUnlessTargetIsObject(Object.getOwnPropertyDescriptor),
+
+        getPrototypeOf: throwUnlessTargetIsObject(Object.getPrototypeOf),
+
+        has: throwUnlessTargetIsObject(function has(target, key) {
+          return key in target;
+        }),
+
+        isExtensible: throwUnlessTargetIsObject(Object.isExtensible),
+
+        // Basically the result of calling the internal [[OwnPropertyKeys]].
+        // Concatenating propertyNames and propertySymbols should do the trick.
+        // This should continue to work together with a Symbol shim
+        // which overrides Object.getOwnPropertyNames and implements
+        // Object.getOwnPropertySymbols.
+        ownKeys: throwUnlessTargetIsObject(function ownKeys(target) {
+          var keys = Object.getOwnPropertyNames(target);
+
+          if (ES.IsCallable(Object.getOwnPropertySymbols)) {
+            keys = keys.concat(Object.getOwnPropertySymbols(target));
```

Comment:
Would this be faster as `keys.push.apply(keys, Object.getOwnPropertySymbols(target))`?